internal/consensus/common_test.go

```diff
@@ -507,18 +507,33 @@ func loadPrivValidator(t *testing.T, cfg *config.Config) *privval.FilePV {
 	return privValidator
 }
 
-func makeState(ctx context.Context, t *testing.T, cfg *config.Config, logger log.Logger, nValidators int) (*State, []*validatorStub) {
+type makeStateArgs struct {
+	validators  int
+	application abci.Application
+}
+
+func makeState(ctx context.Context, t *testing.T, cfg *config.Config, logger log.Logger, args makeStateArgs) (*State, []*validatorStub) {
```

Comment:
It's not crucial, but is there some reason why cfg and logger aren't also in the args type?

Reply:
Not a strong one. Mostly that I wasn't sure it made sense to ever provide defaults since most tests use them in multiple places. Not a great reason, so switched.